flink-table-store-hive/flink-table-store-hive-catalog/src/main/java/org/apache/flink/table/store/hive/HiveCatalog.java

```diff
@@ -269,6 +271,26 @@ protected String warehouse() {
         return hiveConf.get(HiveConf.ConfVars.METASTOREWAREHOUSE.varname);
     }
 
+    private void checkObjectPathUpperCase(ObjectPath objectPath) {
+        checkState(
+                objectPath.getDatabaseName().equals(objectPath.getDatabaseName().toLowerCase()),
+                String.format(
+                        "Database name[%s] cannot contain upper case",
+                        objectPath.getDatabaseName()));
+        checkState(
+                objectPath.getObjectName().equals(objectPath.getObjectName().toLowerCase()),
+                String.format(
+                        "Table name[%s] cannot contain upper case", objectPath.getObjectName()));
+    }
+
+    private void checkFieldNamesUpperCase(List<String> fieldNames) {
+        for (String fieldName : fieldNames) {
+            checkState(
```

Comment:
Could we check the state of `fieldNames` not single `fieldName`? IMO, it's better to provide users with all upper case field names once.